lib/autotune-prep/categorize.js

```diff
@@ -152,6 +163,26 @@ function categorizeBGDatums(opts) {
         //sens = ISF
         var sens = ISF.isfLookup(IOBInputs.profile.isfProfile,BGDate);
         IOBInputs.clock=BGDate.toISOString();
+        // trim down IOBInputs.history to just the data for 6h prior to BGDate
+        //console.error(IOBInputs.history[0].created_at);
+        var newHistory = [];
+        for (h=0; h<fullHistory.length; h++) {
+            var hDate = new Date(fullHistory[h].created_at)
+            //console.error(fullHistory[i].created_at, hDate, BGDate, BGDate-hDate);
+            //if (h == 0 || h == fullHistory.length - 1) {
+                //console.error(hDate, BGDate, hDate-BGDate)
+            //}
+            if (BGDate-hDate < 6*60*60*1000 && BGDate-hDate > 0) {
```

Comment:
i'm not sure how likely it is, or if it affects the result, but should this timeout be max(dia, 6h)?

Reply:
To do that would require some re-plumbing to get DIA into this part of the code.  Even for people with 7h or 8h DIA, I'm not sure that including the older insulin treatments would be any more accurate, since we stop considering carbs after 6h.  And either way, it'd be a tiny effect, so I'm fine leaving it as is with 6h hard-coded, unless someone else wants to do the work to make it use DIA.